ucp/_libs/ucx_api.pyx

```diff
@@ -618,14 +618,14 @@ cdef class UCXRequest:
             )
 
 
-cdef _handle_status(
+cdef UCXRequest _handle_status(
     ucs_status_ptr_t status,
     int64_t expected_receive,
     cb_func,
     cb_args,
     cb_kwargs,
-    name,
-    inflight_msgs
+    unicode name,
+    set inflight_msgs
```

Comment:
I see that making this a `set` matches other entries in the file. I would though expect something named as inflight messages to be a `list` instead. Wouldn't it make more sense? Isn't there a chance that you can have the same message more than once from multiple calls?

Reply:
The definition comes from [here]( https://github.com/rapidsai/ucx-py/blob/21d6d5cca9fb6b826a87d46cb231cc5a9c43ff5f/ucp/_libs/ucx_api.pyx#L293 ). All we are doing in this PR is making sure it is typed correctly so Cython can use [more efficient C APIs with the `set`]( https://docs.python.org/3/c-api/set.html ). Maybe it would be better to ask this question in a new issue?

Reply:
Yeah, I thought that would be overstepping this PR's context, just making sure. Thanks for clarifying @jakirkham .

Reply:
>  Wouldn't it make more sense? Isn't there a chance that you can have the same message more than once from multiple calls?

Not as far as I know, `inflight_msgs` contains `UCXRequest`, which [hashes to the UID of the underlying `ucx_py_request`](https://github.com/rapidsai/ucx-py/blob/21d6d5cca9fb6b826a87d46cb231cc5a9c43ff5f/ucp/_libs/ucx_api.pyx#L602).